src/main/scala/symsim/examples/concrete/cartpole/CartPole.scala

```diff
@@ -0,0 +1,171 @@
+      val dcp = closest (s.cp) (cPCutPoints)
+      val dcv = closest (s.cv) (cVCutPoints)
+      val dpa = closest (s.pa) (pACutPoints)
+      val dpv = closest (s.pv) (pVCutPoints)
+
+      CartPoleState (dcp, dcv, dpa, dpv)
+
+
+    private def cartPoleReward (s: CartPoleState) (a: CartPoleAction): CartPoleReward =
+      1
+
+
+    // TODO: this is now deterministic but eventually needs to be randomized
+    def step (s: CartPoleState) (a: CartPoleAction): Randomized[(CartPoleState, CartPoleReward)] =
+      require (instances.enumAction.membersAscending.contains (a))
+      val costheta = Math.cos(s.pa)
+      val sintheta = Math.sin(s.pa)
+
+      val temp = (
+        a * force_mag + polemass_length * Math.pow(s.pv, 2) * sintheta
+        ) / total_mass
+      val thetaacc = (gravity * sintheta - costheta * temp) / (
+        length * (4.0 / 3.0 - masspole * Math.pow(costheta, 2) / total_mass)
+        )
+      val xacc = temp - polemass_length * thetaacc * costheta / total_mass
+
+      val cp1 = s.cp + tau * s.cv
+      val cv1 = s.cv + tau * xacc
+      val pa1 = s.pa + tau * s.pv
+      val pv1 = s.pv + tau * thetaacc
+
+
+      val s1 = CartPoleState (cp = cp1, cv = cv1, pa = pa1, pv = pv1)
+      Randomized.const (s1, cartPoleReward (s1) (a))
+
+
+    def initialize: Randomized[CartPoleState] =
+      for
+        cp <- Randomized.repeat (Randomized.between (-0.05, 0.05))
+        cv <- Randomized.repeat (Randomized.between (-0.05, 0.05))
+        pa <- Randomized.repeat (Randomized.between (-0.05, 0.05))
+        pv <- Randomized.between(-0.05, 0.05)
+        s = CartPoleState (cp, cv, pa, pv) if !isFinal (s)
+      yield s
+
+end CartPole
+
+
+/** Here is a proof that our types actually deliver on everything that an Agent
+  * needs to be able to do to work in the framework.
+  */
+object CartPoleInstances
+  extends AgentConstraints[CartPoleState, CartPoleObservableState, CartPoleAction, CartPoleReward, Randomized]:
+
+  import cats.{Eq, Monad, Foldable}
+  import cats.kernel.BoundedEnumerable
+
+  import org.scalacheck.Gen
+  import org.scalacheck.Arbitrary
+  import org.scalacheck.Arbitrary.arbitrary
+
+  given enumAction: BoundedEnumerable[CartPoleAction] =
+    BoundedEnumerableFromList (-1, 1)
+
+  given enumState: BoundedEnumerable[CartPoleObservableState] =
+    val ss: List[CartPoleObservableState] = for
+      cp <- cPCutPoints
+      cv <- cVCutPoints
+      pa <- pACutPoints
+      pv <- pVCutPoints
+    yield CartPoleState (cp, cv, pa, pv)
+    BoundedEnumerableFromList(ss *)
+
+  given schedulerIsMonad: Monad[Randomized] =
+    concrete.Randomized.randomizedIsMonad
+
+  given schedulerIsFoldable: Foldable[Randomized] =
+    concrete.Randomized.randomizedIsFoldable
+
+  given canTestInScheduler: CanTestIn[Randomized] =
+    concrete.Randomized.canTestInRandomized
+
+  lazy val genCartPoleState: Gen[CartPoleState] = for
+    cp <- Gen.choose[Double] (CP_MIN, CP_MAX)
+    cv <- Gen.choose[Double] (CV_MIN, CV_MIN)
```

Comment:
@mohsen-ghaffari1992, why is this `Gen.choose` using `CV_MIN` for both parameters? Should  the other one be `CV_MAX`?

Reply:
@wasowski 
Sure. It is a typo.

Reply:
thanks I will fix it.